code/go/0chain.net/smartcontract/dbs/event/eventdb.go

```diff
@@ -15,17 +20,81 @@ func NewEventDb(config dbs.DbAccess) (*EventDb, error) {
 		return nil, err
 	}
 	eventDb := &EventDb{
-		Store: db,
+		Store:          db,
+		eBufferChannel: make(chan eventCtx, 100),
+		eChannel:       make(chan eventCtx, 100),
 	}
 
 	if err := eventDb.AutoMigrate(); err != nil {
 		return nil, err
 	}
+	go eventDb.channelBufferIntermediate()
+	go eventDb.addEventWorker()
 	return eventDb, nil
 }
 
+type eventCtx struct {
+	ctx    context.Context
+	events []Event
+}
+
 type EventDb struct {
 	dbs.Store
+	eChannel       chan eventCtx
+	eBufferChannel chan eventCtx
+}
+
+// helps maitaining dynamic buffer for addEventWorker
+func (edb EventDb) channelBufferIntermediate() {
+	buf := make([]eventCtx, 0)
+	for {
+		events, ok := <-edb.eBufferChannel
+		for _, e := range buf {
+			select {
+			case edb.eChannel <- e:
+			default:
+				break
+			}
+			buf = buf[1:]
```

Comment:
I don't think that it is a good idea to drop events without any notification here, at least logging